### ISSUE TITLE
[BUG]flutter: scheduleTask ERROR: PlatformException

### ISSUE THREAD

**Issue by vifird** [transistorsoft/flutter_background_fetch#84]
**Your Environment**
* Plugin version: 0.5.4
* Platform: iOS
* OS version: 10.15.2
* Device manufacturer / model: iPhone XS
* Flutter info (`flutter info`, `flutter doctor`): flutter 
```
Flutter 1.15.3 • channel dev • https://github.com/flutter/flutter.git
Framework • revision 67826bdce5 (2 weeks ago) • 2020-02-10 14:59:32 -0800
Engine • revision 6158f03ef5
Tools • Dart 2.8.0 (build 2.8.0-dev.8.0 514a8d4c84)
```
* Plugin config

**To Reproduce**
1. Config as guide in xcode.
2. Write my code:
```dart
Future<void> configBackgroundTask() async {
  debugPrint('>>[BackgroundFetch] configBackgroundTask');
  // Configure BackgroundFetch
  await BackgroundFetch.configure(
      BackgroundFetchConfig(
        minimumFetchInterval: 15,
        stopOnTerminate: false,
        enableHeadless: true,
        requiresBatteryNotLow: false,
        requiresCharging: false,
        requiresStorageNotLow: false,
        requiresDeviceIdle: false,
        requiredNetworkType: NetworkType.ANY,
      ), (String taskId) {
    debugPrint('>>[BackgroundFetch] configBackgroundTask taskId: $taskId');
    _startHeadlessTask(taskId);
  }).then((status) {
    debugPrint('>>[BackgroundFetch] configure success: $status');
  }, onError: (e) {
    print('>>[BackgroundFetch] configure ERROR: $e');
  });

  await BackgroundFetch.scheduleTask(
    TaskConfig(
      taskId: "com.transistorsoft.fetch",
      delay: 5000, // <-- milliseconds
    ),
  ).then((value) {
    debugPrint('>>[BackgroundFetch] scheduleTask success: $value');
  }, onError: (e) {
    print('>>[BackgroundFetch] scheduleTask ERROR: ${e.toString()}');
    throw e;
  });

  BackgroundFetch.registerHeadlessTask(_startHeadlessTask);
}

```
3. Start, and scheduleTask cannot run correctly, throw error: `scheduleTask ERROR: PlatformException(0, Background procssing task was not registered in AppDelegate didFinishLaunchingWithOptions.  See iOS Setup Guide., null)`

**Debug logs**
```
flutter: >>[BackgroundFetch] configBackgroundTask
flutter: >>[BackgroundFetch] configure success: 2
flutter: >>[BackgroundFetch] scheduleTask ERROR: PlatformException(0, Background procssing task was not registered in AppDelegate didFinishLaunchingWithOptions.  See iOS Setup Guide., null)
```

**Additional context**
Here is my `AppDelegate.swift`
```swift
import UIKit
import Flutter

@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    GeneratedPluginRegistrant.register(with: self)
    // flutter_local_notification config
    if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
    }
    if(!UserDefaults.standard.bool(forKey: "Notification")) {
      UIApplication.shared.cancelAllLocalNotifications()
      UserDefaults.standard.set(true, forKey: "Notification")
    }
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
}
```

**Comment by christocracy** [comment]
Did you do as the error suggested and "See iOS Setup Guide"?

**Comment by christocracy** [comment]
You can't use `scheduleTask` on iOS without adding *your own* custom `taskId`, eg: `com.transistorsoft.MyCustomTaskId`.

`com.transistorsoft.fetch` is reserved for the fetch-task.

Make up your own taskId and add it to your .plist as documented in the *iOS Setup*, linked in the README.

**Comment by vifird** [comment]
@christocracy Got it, so we need add two taskId if using 'scheduleTask'. Thanks a lot.

**Comment by uc-dve** [comment]
@christocracy 
I have added two task id but still facing same issue. I have tried whole day but nothing happened. Please suggest me if i am missing something
![image](https://user-images.githubusercontent.com/28289763/106131486-bed4dd80-6188-11eb-9236-b8437ad61052.png)

here is my AppDelegate.swift code

```
import UIKit
import Flutter
import Firebase

@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    FirebaseApp.configure()
    //  if #available(iOS 10.0, *) {
    //   UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
    // }
    GeneratedPluginRegistrant.register(with: self)
    UNUserNotificationCenter.current().delegate = self
    UIApplication.shared.setMinimumBackgroundFetchInterval(UIApplication.backgroundFetchIntervalMinimum)
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
  
}
```

**Comment by christocracy** [comment]
>UIApplication.shared.setMinimumBackgroundFetchInterval(UIApplication.backgroundFetchIntervalMinimum)

Why are you doing this manually?  **Delete that line**.  `flutter_background_fetch` does this automatically.  Also, that is the [old *deprecated way* of enabling iOS fetch events](https://developer.apple.com/documentation/uikit/uiapplication/1623100-setminimumbackgroundfetchinterva?language=objc).  By using that line, you are disabling the new [`BGAppRefeshTask`](https://developer.apple.com/documentation/backgroundtasks/bgapprefreshtask?language=objc) API.

Here is the [`AppDelegate`](https://github.com/transistorsoft/flutter_background_fetch/blob/master/example/ios/Runner/AppDelegate.m) of the `/example` app in this repo.  I suggest you clone this repo and boot the `/example` app.

**Comment by christocracy** [comment]
[here](https://github.com/transistorsoft/transistor-background-fetch/blob/master/ios/TSBackgroundFetch/TSBackgroundFetch/TSBackgroundFetch.m#L93-L107) is how the SDK *automatically* executes `setMinimumBackgroundFetchInterval`, by first checking if the device is running `iOS 13+`.

```obj-c
- (NSError*) scheduleBGAppRefresh {
    if (fetchScheduled) return nil;
    
    NSLog(@"[%@ scheduleBGAppRefresh] %@", TAG, BACKGROUND_REFRESH_TASK_ID);
        
    NSError *error = nil;
    
    if (@available (iOS 13.0, *)) {
        if ([TSBGAppRefreshSubscriber useTaskScheduler]) {
            BGTaskScheduler *scheduler = [BGTaskScheduler sharedScheduler];
            BGAppRefreshTaskRequest *request = [[BGAppRefreshTaskRequest alloc] initWithIdentifier:BACKGROUND_REFRESH_TASK_ID];
            request.earliestBeginDate = [NSDate dateWithTimeIntervalSinceNow:minimumFetchInterval];
            [scheduler submitTaskRequest:request error:&error];
        } else {
            [self setMinimumFetchInterval];
        }
    } else {
        [self setMinimumFetchInterval];
    }
    if (!error) {
        fetchScheduled = YES;
    }
    return error;
}
```

**Comment by Acventor** [comment]
> @christocracy I have added two task id but still facing same issue. I have tried whole day but nothing happened. Please suggest me if i am missing something ![image](https://user-images.githubusercontent.com/28289763/106131486-bed4dd80-6188-11eb-9236-b8437ad61052.png)
> 
> here is my AppDelegate.swift code
> 
> ```
> import UIKit
> import Flutter
> import Firebase
> 
> @UIApplicationMain
> @objc class AppDelegate: FlutterAppDelegate {
>   override func application(
>     _ application: UIApplication,
>     didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
>   ) -> Bool {
>     FirebaseApp.configure()
>     //  if #available(iOS 10.0, *) {
>     //   UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
>     // }
>     GeneratedPluginRegistrant.register(with: self)
>     UNUserNotificationCenter.current().delegate = self
>     UIApplication.shared.setMinimumBackgroundFetchInterval(UIApplication.backgroundFetchIntervalMinimum)
>     return super.application(application, didFinishLaunchingWithOptions: launchOptions)
>   }
>   
> }
> ```
> 
> issue screenshot -
> 
> ![image](https://user-images.githubusercontent.com/28289763/106132096-7ff35780-6189-11eb-8e4e-1d060826a259.png)

This is a late response, but here is the solution - if your `Info.plist` is splitted into two parts `release` and `debug`, then you should do setup for both of them.
